File: perl/002/002.pl
use strict;

use List::Util qw(reduce);

my @fibs = (0, 1);

while (@fibs[-1] + @fibs[-2] < 4000000) {
  push(@fibs, @fibs[-1] + @fibs[-2]);
}

print reduce { $a + $b } grep($_ % 2 == 0, @fibs);

print "\n";

File: javascript/002/002.js
"use strict";

let fibs = [1, 2];

while (fibs.slice(-1)[0] < 4000000) {
  fibs.push(fibs.slice(-2)[0] + fibs.slice(-1)[0]);
}

console.log(fibs.filter((element) => element % 2 === 0).reduce((previousValue, currentValue) => previousValue + currentValue));
